lockfile/lockfile-file/src/sortLockfileKeys.ts

```diff
@@ -39,14 +39,15 @@ const ROOT_KEYS_ORDER = {
   onlyBuiltDependencies: 3,
   overrides: 4,
   packageExtensionsChecksum: 5,
-  patchedDependencies: 6,
-  specifiers: 10,
-  dependencies: 11,
-  optionalDependencies: 12,
-  devDependencies: 13,
-  dependenciesMeta: 14,
-  importers: 15,
-  packages: 16,
+  pnpmfileChecksum: 6,
+  patchedDependencies: 7,
+  specifiers: 11,
+  dependencies: 12,
+  optionalDependencies: 13,
+  devDependencies: 14,
+  dependenciesMeta: 15,
+  importers: 16,
+  packages: 17,
```

Comment:
it is not necessary to bump the numbers for these. 10 was still bigger than 7

Reply:
```suggestion
  specifiers: 10,
  dependencies: 11,
  optionalDependencies: 12,
  devDependencies: 13,
  dependenciesMeta: 14,
  importers: 15,
  packages: 16,
```

Reply:
Why wasn't it an array, though? It should be easy to use code to convert an array to `Record<string, number>`.

Reply:
It is more optimized this way. In an array you'd have to search for the item on each lockfile entry.

Reply:
> In an array you'd have to search for the item on each lockfile entry.

By "search", I assume you mean searching programmatically (instead of searching in the editor). This would indeed be inefficient. But you can convert the array to object right away:

```typescript
const ROOT_KEYS: ReadonlyArray<keyof typeof Lockfile> = [ 'specifiers', 'dependencies', '...' ]
const ROOT_KEY_ORDER: Record<keyof typeof Lockfile, number> = Object.fromEntries(ROOT_KEYS.map((key, index) => [key, index]))
```

Reply:
That would be fine probably. The additional processing is negligible.

Reply:
@zkochan https://github.com/pnpm/pnpm/pull/7686